### DISCUSSION TITLE
Autotuner Use

### DISCUSSION THREAD

**Discussion by Wiki1998-dev** [The-OpenROAD-Project/OpenROAD-flow-scripts#2413] [Q&A] [answered]
Hi , 
i am trying to optimize the parameters of my design  using autotuner 
few question i had to ask 
we open the python script destributed.py from util folder then why is there one also in tools autotuner src 
secondly our aim is Power performance area  optimisation right ?
where can i see that graph ?or should i make it specifically myself
thirdly what and how to see the tunnnnnnning annd optimisation curve?

**Comment by Wiki1998-dev** [comment]
https://drive.google.com/file/d/1D5dLMFp4M3P45zHxLzsLjFnuwxH8EHNQ/view?usp=drive_link
this is the design with all the files which may be needed

**Comment by maliberty** [comment]
@luarss please help.

**Comment by luarss** [comment, marked as answer]
Welcome to OpenROAD!

> we open the python script destributed.py from util folder then why is there one also in tools autotuner src

You have to use the one in `./tools/AutoTuner/src`.  You might have to install the prerequisites first. Please refer to the docs: https://openroad-flow-scripts.readthedocs.io/en/latest/user/InstructionsForAutoTuner.html#setting-up-autotuner

Example command:
```
cd tools/AutoTuner/src/autotuner
python distributed.py --design gcd --platform asap7 --config ../../../../flow/designs/asap7/gcd/autotuner.json tune --samples 5
```

> secondly our aim is Power performance area optimisation right ?

The current cost function is based on `F_max` and `num_drcs`. You can implement any cost function to achieve your goals. See the evaluation function here: https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/cf878af0cdf8b5af164c509707cdcf590caf1503/tools/AutoTuner/src/autotuner/distributed.py#L98

> where can i see that graph ?or should i make it specifically myself
thirdly what and how to see the tunnnnnnning annd optimisation curve?

We do not generate any graph automatically. However, we support TensorBoard and you can check out the docs here: https://openroad-flow-scripts.readthedocs.io/en/latest/user/InstructionsForAutoTuner.html#gui

Finally, does the design completes the ORFS flow as is? If it does not, the first step is to follow the documentation on adding new designs: https://openroad-flow-scripts.readthedocs.io/en/latest/user/AddingNewDesign.html#adding-new-designs-to-the-orfs
